### PR TITLE
[BUGFIX] Prevent php warning when passing array as tag attribute

### PR DESCRIPTION
**Pull request by s2b**
TagBuilder::addAttribute() allows both strings and array/traversables as attribute value. However, in the implementation only "data" and "aria" are properly handled as arrays. For other attribute names, this leads to a type conversion warning.

With this patch, that edge case is properly handled with an exception. Also, test coverage is added both for data/aria and the exception.

Resolves: #857

### CODE REVIEW

File: src/Core/ViewHelper/TagBuilder.php [hunk 1]
```diff
@@ -190,9 +193,14 @@ public function addAttribute($attributeName, $attributeValue, $escapeSpecialChar
         if ($escapeSpecialCharacters) {
             $attributeName = htmlspecialchars($attributeName);
         }
-        if (in_array($attributeName, ['data', 'aria'], true)
-            && (is_array($attributeValue) || $attributeValue instanceof \Traversable)
-        ) {
+        if (is_array($attributeValue) || $attributeValue instanceof \Traversable) {
+            if (!in_array($attributeName, ['data', 'aria'], true)) {
+                throw new \InvalidArgumentException(
+                    sprintf('Specified tag attribute "%s" cannot be of type array.', $attributeName),
```
Comment (liayn) [inline src/Core/ViewHelper/TagBuilder.php:199]: wording suggestion: "Value of tag attribute %s can't be of type array"

Comment (s2b) [reply src/Core/ViewHelper/TagBuilder.php]: fixed

**Review by lolli42 — APPROVED** [review]
